pallets/msa/src/lib.rs

```diff
@@ -131,17 +154,20 @@ impl<T: Config> Pallet<T> {
 	}
 
 	pub fn add_key(msa_id: MessageSenderId, key: &T::AccountId) -> DispatchResult {
-		<KeyOwnerOf<T>>::try_mutate(key, |maybe_msa| {
+		<KeyInfoOf<T>>::try_mutate(key, |maybe_msa| {
 			ensure!(maybe_msa.is_none(), Error::<T>::DuplicatedKey);
 
-			*maybe_msa = Some(msa_id);
+			*maybe_msa =
+				Some(KeyInfo { msa_id, expired: T::BlockNumber::zero(), nonce: Zero::zero() });
```

Comment:
how is the nonce to be used?  It's not being incremented anywhere.

Reply:
You are right. It was a suggestion made during planning.

Reply:
Nonce is to make the data which is going to be signed unique, so nobody can use that signature for something similar. Nonce should not get stored in here as far as I know.

Reply:
I think there are great reasons to added but it was added a bit too soon.